test/moq_message.cpp

```diff
@@ -0,0 +1,492 @@
+    buffer << uintVar_t(0x2);
+    quicr::bytes payload_2{9,9,9,9,9};
+    buffer << payload_2;
+
+
+
+    uint8_t msg_type {0};
+    buffer >> msg_type;
+    CHECK_EQ(msg_type, MESSAGE_TYPE_STREAM_HEADER_TRACK);
+
+
+    MoqStreamHeaderTrack sth_out;
+    buffer >> sth_out;
+    CHECK_EQ(sth.track_alias, sth_out.track_alias);
+    CHECK_EQ(sth.subscribe_id, sth_out.subscribe_id);
+    CHECK_EQ(sth.priority, sth_out.priority);
+
+    uintVar_t group_id{0};
+    uintVar_t object_id{0};
+    quicr::bytes payload{};
+
+    // object 1
+    buffer >> group_id;
+    buffer >> object_id;
+    buffer >> payload;
+
+    CHECK_EQ(0x100, group_id);
+    CHECK_EQ(0x1, object_id);
+    CHECK_EQ(payload_1, payload);
+
+
+    // object 2
+    buffer >> group_id;
+    buffer >> object_id;
+    buffer >> payload;
+
+    CHECK_EQ(0x100, group_id);
+    CHECK_EQ(0x2, object_id);
+    CHECK_EQ(payload_2, payload);
+}
+
+
+TEST_CASE("MultiObjectStream StreamHeader Group Format encode/decode")
+{
+    auto stg = MoqStreamHeaderGroup {
+      .subscribe_id = 0xABCD,
+      .track_alias = TRACK_ALIAS_ALICE_VIDEO,
+      .group_id = 0x100,
+      .priority = 0xC
+    };
+
+    MessageBuffer buffer;
+    buffer << stg;
+    // add 2 objects
+    buffer << uintVar_t(0x1);
+    quicr::bytes payload_1{1,2,3,4,5};
+    buffer << payload_1;
+
+    buffer << uintVar_t(0x2);
+    quicr::bytes payload_2{9,9,9,9,9};
+    buffer << payload_2;
+
+
+
+    uint8_t msg_type {0};
+    buffer >> msg_type;
+    CHECK_EQ(msg_type, MESSAGE_TYPE_STREAM_HEADER_GROUP);
+
+
+    MoqStreamHeaderGroup stg_out;
+    buffer >> stg_out;
+    CHECK_EQ(stg.track_alias, stg_out.track_alias);
+    CHECK_EQ(stg.subscribe_id, stg_out.subscribe_id);
+    CHECK_EQ(stg.group_id, stg_out.group_id);
+    CHECK_EQ(stg.priority, stg_out.priority);
+
+    uintVar_t object_id{0};
+    quicr::bytes payload{};
+
+    // object 1
+    buffer >> object_id;
+    buffer >> payload;
+    CHECK_EQ(0x1, object_id);
+    CHECK_EQ(payload_1, payload);
+
+
+    // object 2
+    buffer >> object_id;
+    buffer >> payload;
+    CHECK_EQ(0x2, object_id);
+    CHECK_EQ(payload_2, payload);
+}
+
+/// Some trivial encode and decode for QUIC Varint
+struct A {
+  quicr::uintVar_t val;
+};
+
+MessageBuffer &
+operator<<(MessageBuffer &msg, const struct A &value) {
+    msg << value.val;
+    return msg;
+}
+
+MessageBuffer &
+operator>>(MessageBuffer &msg, struct A &value) {
+    struct A a {};
+    msg >> a.val;
+    value.val = a.val;
+    return msg;
+}
+
+TEST_CASE("QUIC Varint") {
+    A a;
+    a.val = 15293;
+
+    MessageBuffer buffer;
+    buffer << a;
+    A a_out;
+    buffer >> a_out;
+    CHECK_EQ(a.val, a_out.val);
+
+    //.start_group = Location { .mode = LocationMode::Absolute, .value = 100},
+    Location l_in = {
+       .mode = LocationMode::Absolute,
+       .value = 100,
+    };
+
+    buffer << l_in;
+    Location l_out;
+    buffer >> l_out;
+    CHECK_EQ(l_in.mode, l_out.mode);
```

Comment:
Who is this test for? `A`? Or `Location`? THe name says it is for Varint, but that's been obfuscated away by wrapping it other types.